### DISCUSSION TITLE
How to operate the secondary development based on the source code

### DISCUSSION THREAD

**Discussion by GracefulTing** [coder/code-server#3499] [Q&A]
Does it only support downloading binary files to run? What if I want to do secondary development based on source code? How to develop new functions on this basis and debug locally?

**Comment by code-asher** [comment]
This file should contain everything you need to develop code-server: https://github.com/cdr/code-server/blob/main/docs/CONTRIBUTING.md#requirements

I see you posted an error here https://github.com/cdr/code-server/discussions/3488#discussioncomment-795994 so let's continue the discussion there. :)
